File: app/src/main/java/com/clockmods/background/BackgroundDimSchedule.java
package com.clockmods.background;

public final class BackgroundDimSchedule {
    private BackgroundDimSchedule() {
    }

    public static boolean isActive(int currentMinutes, int startMinutes, int endMinutes) {
        if (startMinutes == endMinutes) {
            return false;
        }
        if (startMinutes < endMinutes) {
            return currentMinutes >= startMinutes && currentMinutes < endMinutes;
        }
        return currentMinutes >= startMinutes || currentMinutes < endMinutes;
    }
}

File: app/src/main/java/com/clockmods/background/BackgroundDimSchedule.kt
package com.clockmods.background

object BackgroundDimSchedule {
    @JvmStatic
    fun isActive(currentMinutes: Int, startMinutes: Int, endMinutes: Int): Boolean {
        if (startMinutes == endMinutes) return false
        return if (startMinutes < endMinutes) {
            currentMinutes >= startMinutes && currentMinutes < endMinutes
        } else {
            currentMinutes >= startMinutes || currentMinutes < endMinutes
        }
    }
}
